core/transfer/src/main/java/org/eclipse/dataspaceconnector/transfer/core/transfer/TransferProcessManagerImpl.java

```diff
@@ -517,6 +518,18 @@ private void handleDeprovisionResult(TransferProcess transferProcess, List<Depro
         }
     }
 
+    private void removeDeprovisionedSecrets(ProvisionedDataAddressResource provisionedResource, TransferProcess transferProcess) {
+        if (!provisionedResource.hasToken()) {
```

Comment:
this guard condition could be moved to the condition at line 499 to improve clarity.

Reply:
+1